core/src/components/toast/toast.tsx

```diff
@@ -315,7 +349,16 @@ export class Toast implements ComponentInterface, OverlayInterface {
       'toastLeave',
       iosLeaveAnimation,
       mdLeaveAnimation,
-      this.position
+      /**
+       * Fetch the cached position that was calculated back in the present
+       * animation. We always want to animate the dismiss from the same
+       * position the present stopped at, so the animation looks continuous.
+       */
+      {
+        position,
+        top: lastPresentedPosition?.top ?? '',
+        bottom: lastPresentedPosition?.bottom ?? '',
+      }
```

Comment:
I know `lastPresentedPosition` is updated whenever `present` is called again, but can we still clear the reference in `dismiss`? The information is no longer relevant after the toast has been dismissed.